### PR TITLE
HOSTEDCP-1791: dont render cloud-credentials secret

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/util/fixture.go [hunk 1]
```diff
@@ -183,6 +183,7 @@ func renderCreate(ctx context.Context, opts *core.RawCreateOptions, platformOpts
 
 	opts.Render = true
 	opts.RenderInto = outputFile
+	opts.RenderSensitive = false
```
Comment (fxierh) [inline test/e2e/util/fixture.go:186]: Will this get overridden in core.CreateCluster below ?

Comment (Patryk-Stefanski) [reply test/e2e/util/fixture.go:186]: It was by the complete function but it should anymore.

File: cmd/cluster/core/create.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ func bindCoreOptions(opts *RawCreateOptions, flags *pflag.FlagSet) {
 	flags.StringVar(&opts.ControlPlaneAvailabilityPolicy, "control-plane-availability-policy", opts.ControlPlaneAvailabilityPolicy, "Availability policy for hosted cluster components. Supported options: SingleReplica, HighlyAvailable")
 	flags.BoolVar(&opts.Render, "render", opts.Render, "Render output as YAML to stdout instead of applying")
 	flags.StringVar(&opts.RenderInto, "render-into", opts.RenderInto, "Render output as YAML into this file instead of applying. If unset, YAML will be output to stdout.")
+	flags.BoolVar(&opts.RenderSensitive, "render-sensitive", opts.RenderSensitive, "Render sensitive information in the output")
```
Comment (fxierh) [inline cmd/cluster/core/create.go:73]: It looks like RenderSensitive defaults to false now. Any chance of defaulting this to true so it won't break existing jobs which make use of the create - patch - apply logic ? 

Maybe we could negate the flag by making it `--no-render-sensitive`.

Comment (Patryk-Stefanski) [reply cmd/cluster/core/create.go:73]: Where does it default to false, from my understanding the is no default value and we only set it to false for azure e2e.

Comment (fxierh) [reply cmd/cluster/core/create.go:73]: It is a bool so its value will be false unless we set it to true

**Comment by enxebre** [comment]
Please default to not rendering sensitive info once the relevant jobs have been updated.
fwiw this won't be an issue when moving to msi
/approve

**Comment by bryan-cox** [comment]
/retest

for konflux

**Comment by stevekuznetsov** [comment]
>  Please default to not rendering sensitive info ...

@enxebre wouldn't this break every consumer of `--render` that uses the output as a self-contained mechanism to install `HostedClusters` on a management plane?

**Pull request merged**